### ISSUE TITLE
[research question] does the model predict the last few codes in delayed pattern

### ISSUE THREAD

**Issue by jasonppy** [facebookresearch/audiocraft#95]
Hi,

Thanks for the great work and open sourcing everything!

On the delayed pattern
<img width="612" alt="image" src="https://github.com/facebookresearch/audiocraft/assets/47729801/5377b218-ec77-4a9c-a403-32457d5bd1f5">

it seems that there should be a few model tokens at the end to be predicted (I drew some of them out on the left). Are those predicted? or ignoring those doesn't affect result (which is reasonable considering the first codebook is the most important and only a very small amount of codes from higher codebooks are dropped)

Also, Looking forward to the training code!

Thanks

**Comment by chavinlo** [comment]
I've been working on a trainer (first time making one). From what I've seen this happens on the `compute_predictions` function on `model.lm`, where the last layers are NaN in a triangle (?) shape like this:

![Screenshot_20230614-093958_Chrome](https://github.com/facebookresearch/audiocraft/assets/85657083/fa075aa2-7a64-477a-93fe-3945b73c8453)

I just set them back to 0, however my trainer hasn't gotten below 1.3 loss 🥲
The model does seem to generate those logits and I still can't understand why they hid them...

Here are the parts that cause it btw:

https://github.com/facebookresearch/audiocraft/blob/main/audiocraft/modules/codebooks_patterns.py#L264

https://github.com/facebookresearch/audiocraft/blob/main/audiocraft/models/lm.py#L303
